ext/intl/dateformat/dateformat_format_object.cpp

```diff
@@ -101,28 +101,54 @@ U_CFUNC PHP_FUNCTION(datefmt_format_object)
 		}
 
 		idx = 0;
-		while (idx < ht->nNumUsed) {
-			z = &ht->arData[idx].val;
-			if (Z_TYPE_P(z) != IS_UNDEF) {
-				break;
+		if (HT_IS_PACKED(ht)) {
```

Comment:
Not familiar with the history of this file: Is datefmt_format_object performance critical in any web frameworks? If it isn't, then this code is more verbose than it needed to be, even before the refactoring.

If it isn't, using the ZEND_HASH_FOREACH_VAL macro and incrementing idx for each element (and throwing if the final element count wasn't 2, or valid_format was false for any value) could be done instead as an example of how code could be written in the future.

(assigning to dataStyle for idx 0, assigning to timeStyle for idx 1)

Looks correct, though